### ISSUE TITLE
Question about intrisic rank in Figure3(a)

### ISSUE THREAD

**Issue by JulioZhao97** [THU-MIG/yolov10#423]
Hello, thanks for your exellent work! Could you please share the code about how you calculate the intrisic rank in Figure3(a)? I cannot achieve similar results.

**Comment by declanwucv** [comment]
hi @JulioZhao97 , have you get similar results now? I am also trying to calculate the intrinsic rank.

**Comment by JulioZhao97** [comment]
@declanwucv I used the code in this paper: https://arxiv.org/abs/2206.06072, which is mentioned in YOLOv10 paper, and I analysed the rank behavior of COCO trained model and document image trained model, as follows:
![image](https://github.com/user-attachments/assets/e8a66b3c-1fb5-4b5c-a353-03de4116a284)

**Comment by declanwucv** [comment]
Thank you. I just follow the descriptions from the paper, leading to a shape-like result, but the actual values are not even close. Here are my 8 stages rank, the 8, 7, 4 has the lowest rank.
0.21875
0.234375
0.21875
0.07421875
0.1484375
0.171875
0.09375
0.03125

**Comment by JulioZhao97** [comment]
![image](https://github.com/user-attachments/assets/b21389b8-42fe-4eaa-bdcd-6d87456a5740)
The actual values may differ due to input image, but your results are actually very similar? The rank decrease in 1-3, 4 lowest, then 5-6 also decrease and 7-8 lowest

**Comment by minhhotboy9x** [comment]
Hi, I'm also inquiring about intrinsic rank. However, I don't know how to calculate it with conv2d. It has 4 dims so it must be flattened right?

**Comment by lizhehan0905** [comment]
@JulioZhao97 您解决了数值秩的计算方法吗？

**Comment by declanwucv** [comment]
@lizhehan0905 Still I haven't reach that goal yet. But from my personal opinion, the key is the relative value.

**Comment by JulioZhao97** [comment]
@lizhehan0905 No, I can not reproduce the exact values, according to the calculate methods, the values may differ due to input image, preprocess method, and so on, so it is possible to reproduce the exact value, but the similar behavior can be observed.

**Comment by Yipzcc** [comment]
May I ask if you can share the code you calculated?
